### PR TITLE
Adding ability to specify alternative cache

### PR DESCRIPTION
**Pull request by tocco934**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -210,6 +209,10 @@ module.exports = (function() {
 
 module.exports.credential_cache = credential_cache;
 
+module.exports.set_credential_cache = function(cache){
```
Comment (ryanbreen) [inline index.js]: Is this function necessary?  I would think the fact that we export `credential_cache` would mean you could do `cimpress_client_request.credential_cache = {thingy}`

File: test/api_auth.test.js [hunk 2]
```diff
@@ -20,14 +31,15 @@ describe('Auth0 api-auth client grants', function () {
     client_secret: process.env.CIMPRESS_IO_CLIENT_SECRET
   };
 
-  it('Should make a successful request against the backing API', function(done) {
+  it('Should store the access token in the provided cache', function(done) {
```
Comment (ryanbreen) [inline test/api_auth.test.js]: I think you should create this as a new test rather than a replacement of the existing test.

**Comment by ryanbreen** [comment]
Looks good.  Just a couple minor points in review.

File: test/api_auth.test.js (hunk 2 above)
Comment (tocco934) [reply test/api_auth.test.js]: Whoops, I meant to put this logic in the cache.test.js file...I'll move it

**Comment by ryanbreen** [comment]
Can you sync with master and update this PR?  You're missing some changes I pushed to the test cases yesterday, and that's why tests are failing.

**Comment by tocco934** [comment]
Locally it looks like I've successfully pulled your changes and pushing gives me "Everything up-to-date".

This would make more sense if it was just my test that was breaking....

**Comment by ryanbreen** [comment]
Update: I'm an idiot.

**Comment by ryanbreen** [comment]
Merged because tests work locally.  I'll figure out why Travis is being a dork and get this pushed to npm.

**Pull request merged**